test/trace/basic_server.js

```diff
@@ -163,3 +163,17 @@ test('basic tracing test', function (assert) {
         }, 10);
     });
 });
+
+test('modifying traceSample', function (assert){
+    var channel = new TChannel();
+    var subservice = channel.makeSubChannel({
+        serviceName: 'subservice'
+    });
+    assert.equal(channel.traceSample, 0.01);
+    assert.equal(subservice.traceSample, 0.01);
+
+    subservice.setTraceSample(0.99);
+    assert.equal(channel.traceSample, 0.99);
+    assert.equal(subservice.traceSample, 0.99);
```

Comment:
The idea here is that you want a consistent sampling rate across all calls so an entire app instance will use the same sampling rate for all TChannel calls. A call to this interface should apply the samplingRate to the entire channel tree, so it does need to descend into subtrees. It also needs to guard against cycles.